[M2-5716] Participants > Upgrade to Full Account

farmerpaul opened the pull request:
<!-- Use this template as a guide to describe your pull request, and adjust as necessary. -->
<!-- Include information that helps your peers review your updates and understand this    -->
<!-- repository's history of changes over time.                                           -->

<!-- Delete any options that are not relevant -->

- [x] Tests for the changes have been added

### 📝 Description

<!-- Contributions are welcome! If there is a corresponding      -->
<!-- JIRA ticket, link to it by replacing `#` with ticket number -->

🔗 [Jira Ticket M2-5716](https://mindlogger.atlassian.net/browse/M2-5716)

This PR adds the new Upgrade Account popup on the Participants screen, allowing admins to invite Limited accounts to become Full accounts.

- Created `ParticipantSnippet` component that shows secret ID, nickname, and tag in consistent styling (will be used multiple places, including in this popup)
- Created `UpgradeAccountPopup` for the popup itself (reuses existing functionality in `SendInvitationPopup`)
- Integrated the popup into the Participants screen
  - Also updated each participant's `...` menu to show the correct items based on account status, and removed now-unused functionality

> [!TIP]
> Changes have been organized by commit, each with an explanation of what's contained therein; it's recommended to review the PR [commit-by-commit](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1652/commits).

### 📸 Screenshots


https://github.com/ChildMindInstitute/mindlogger-admin/assets/1282772/31159866-7933-42e9-b3c7-8d769be1bfb4


### 🪤 Peer Testing

See Acceptance Criteria of [Jira ticket](https://mindlogger.atlassian.net/browse/M2-5716) for testing instructions.

farmerpaul committed:
M2-5716: fix import circular ref

Fixes runtime compilation error.

farmerpaul committed:
M2-5716: fix regression with empty state

New empty state component was showing on Respondents screen
even when there were respondents listed.

farmerpaul committed:
M2-5716: add participant banner ⇒ shared location

This banner is now used in multiple screens so needs to live under
Dashboard general components. Also needed to move AccountType type to
shared location, so defined in Dashboard general types.

farmerpaul committed:
M2-5716: add shared ParticipantSnippet component

This will be used in a few spots including the account upgrade popup
and some table views, possibly elsewhere. Required body typography style
as well as a palette entry that were found in Figma but missing here.

farmerpaul committed:
M2-5716: add UpgradeAccountPopup

Resuses some functionality from preexisting SendInvitationPopup.

farmerpaul committed:
M2-5716: integrate account upgrade to Participants

Also remove now-unused functionality on Participants screen based on
latest designs.

farmerpaul committed:
M2-5716: revert fill on export sprite

Reverted addition of `fill="currentColor"` on export sprite, as it was
causing menu item icon to be mis-colored. Original styling (before
it was changed in #1593) is actually correct.

farmerpaul committed:
M2-5716: update language strings

farmerpaul committed:
M2-5716: clean up placeholder data

ChaconC commented on src/modules/Dashboard/features/Applet/Popups/UpgradeAccountPopup/UpgradeAccountPopup.tsx:
```diff
@@ -0,0 +1,148 @@
+    if (!appletId || !subjectId) return;
+    Mixpanel.track('Subject Invitation click');
+
+    createInvitation({
+      appletId,
+      subjectId,
+      ...values,
+    });
```
Only issue I'm seeing is the invite email gets sent in English for both language options. For reference, "Add Participant" sends the email in the correct language, but I'm not sure if this is expected here

farmerpaul replied:
You're right, it's missing from the API 😭 I'll add support for it to the API client callback (which thankfully the BE doesn't mind my doing) so that it's strictly a BE-only change.

farmerpaul replied:
Created [M2-6283](https://mindlogger.atlassian.net/browse/M2-6283)

farmerpaul committed:
M2-5716: pass language to account upgrade endpoint

ChaconC reviewed (approved):
Great work!

alex-i-e commented on src/modules/Dashboard/features/Applet/Popups/UpgradeAccountPopup/UpgradeAccountPopup.tsx:
```diff
@@ -0,0 +1,148 @@
+    resolver: yupResolver(UpgradeAccountPopupSchema()),
+    defaultValues,
+    mode: 'all',
+    delayError: 1000,
```
how about a constant value for it? For ex., for input fields we used `CHANGE_DEBOUNCE_VALUE` maybe it applicable in your case as well or does it make sense to introduce a new constant for such form setting?

farmerpaul replied:
I agree, this should be a global constant as it's now used in multiple places, and this aspect of UX should be consistent. Thanks!

farmerpaul committed:
M2-5716: define error delay value as common const

farmerpaul committed:
M2-4823: improve ParticipantSnippet tests

Also extend ParticipantSnippet slightly to align with @sultanofcardio's
parallel changes in #1646.
